gammapy/spectrum/models.py

```diff
@@ -1464,3 +1465,61 @@ def evaluate(self, energy, **kwargs):
         flux = self.spectral_model.evaluate(energy=energy, **kwargs)
         absorption = self.absorption.evaluate(energy=energy, parameter=parameter)
         return flux * absorption
+
+
+class NaimaModel(SpectralModel):
+    r""""""
+    __slots__ = ["radiative_model", "distance"]
+
+    def __init__(self, radiative_model, distance=1.0 * u.kpc):
+        self.radiative_model = radiative_model
+        self.distance = Parameter("distance", distance)
+
+        parameters = []
+        parameters_dict = self.radiative_model.particle_distribution.__dict__
+        for (name, quantity) in parameters_dict.items():
+            if name[0] == "_" or name == "unit":
+                continue
+            parameters.append(Parameter(name, quantity))
```

Comment:
The problem here is that you don't know in principle what are the parameters of the model.. If we want to have this general `NaimaModel` class supporting all Naima models I am not sure how to implement the change you suggest. In other words, I don't see how to make something like this work:
```
naima_model = NaimaModel(some_radiative_model)
naima_model.apha
```
Maybe the `NaimaModel` class could have a long `__slots__` list containing all possible model parameters.. but this doesn't seem an ideal solution

Reply:
Thanks for the explanation, indeed this is a problem. Technically you could use something along the lines of:
```
for par in radiative_model.particle_distribution.param_names:
    setter(self, par, Parameter())
```
Declaring the parameters in `__slots__` is not strictly required by the design, but is basically just what we use to avoid that users can set new attributes after init. It might be OK, to leave this part out for the `NaimaModel` for now and maybe change to a different solution for the new attribute handling later. 

The other solution I see would be to implement a dedicated wrapper for all radiative models, very similar to what is done in https://github.com/zblz/naima/blob/master/naima/sherpa_models.py. Not sure if this is good a option...